src/input/json_to_storage.cc

```diff
@@ -85,7 +223,7 @@ bool JSONPath::get_ref_from_head(string & ref_address)
 
     const Node &ref_node = obj.begin()->second;
     if (ref_node.type() != json_spirit::str_type) {
-        THROW( ExcRefOfWrongType() << EI_ErrorAddress(*this) );
+        THROW( ExcRefOfWrongType() << EI_ErrorAddress(this) );
```

Comment:
Passing pointer is not safe. We probably need to use 'as_string' method just here.